systemtest/src/main/java/io/strimzi/systemtest/MessagingBaseST.java

```diff
@@ -0,0 +1,249 @@
+        cliApiClient = new MsgCliApiClient(new URL("http://" + KAFKA_CLIENTS + "." + clientUrl + ":80"));
+    }
+
+    /**
+     * Simple availability check for kafka cluster
+     * @param producer producer
+     * @param consumer consumer
+     * @param clusterName cluster name
+     */
+    void availabilityTest(AbstractClient producer, AbstractClient consumer, String clusterName) throws InterruptedException, ExecutionException, TimeoutException {
+        availabilityTest(producer, consumer, 100, 20000, clusterName, false, "my-topic", null);
+    }
+
+    /**
+     * Simple availability check for kafk acluster
+     * @param producer producer
+     * @param consumer consumer
+     * @param messageCount message count
+     * @param timeout timeout
+     * @param clusterName cluster name
+     */
+    void availabilityTest(AbstractClient producer, AbstractClient consumer, int messageCount, int timeout, String clusterName) throws InterruptedException, ExecutionException, TimeoutException {
+        availabilityTest(producer, consumer, messageCount, timeout, clusterName, false, "my-topic", null);
+    }
+
+    /**
+     * Simple availability check for kafka cluster
+     * @param producer producer
+     * @param consumer consumer
+     * @param messageCount message count
+     * @param timeout timeout for producer and consumer to be finished
+     * @param clusterName cluster name
+     * @param tlsListener option for tls listener inside kafka cluster
+     * @param topicName topic name
+     * @param user user for tls if it's used for messages
+     */
+    void availabilityTest(AbstractClient producer, AbstractClient consumer, int messageCount, int timeout, String clusterName, boolean tlsListener, String topicName, KafkaUser user) throws InterruptedException, ExecutionException, TimeoutException {
+        sendMessages(producer, messageCount, timeout, clusterName, tlsListener, topicName, user);
+        receiveMessages(consumer, messageCount, timeout, clusterName, tlsListener, topicName, user);
+        assertSentAndReceivedMessages(sent, received);
+    }
+
+    /**
+     * Method for send messages to specific kafka cluster. It uses test-client API for communication with deployed clients inside kubernetes cluster
+     * @param producer producer
+     * @param messageCount messages count
+     * @param timeout timeout for producer to be finished
+     * @param clusterName cluster name
+     * @param tlsListener option for tls listener inside kafka cluster
+     * @param topicName topic name
+     * @param user user for tls if it's used for messages
+     * @return count of send and acknowledged messages
+     */
+    int sendMessages(AbstractClient producer, int messageCount, int timeout, String clusterName, boolean tlsListener, String topicName, KafkaUser user) throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
Client's arguments are different. We can create one method, but it will be hard to read the code I think. Now from method name you know what exactly it's doing. We can cut out some things from these methods, but we should keep both of them. For call only one method we have `availabilityTest` method which combines functionality of send and receive.